apis/clusters/v1beta1/cassandra_webhook.go

```diff
@@ -195,6 +195,11 @@ func (cv *cassandraValidator) ValidateUpdate(ctx context.Context, old runtime.Ob
 		return fmt.Errorf("cannot update immutable fields: %v", err)
 	}
 
+	// ensuring if the cluster is ready for the spec updating
+	if (c.Status.CurrentClusterOperationStatus != models.NoOperation || c.Status.State != models.RunningStatus) && c.Generation != oldCluster.Generation {
+		return models.ErrClusterIsNotReadyToUpdate
+	}
+
```

Comment:
Do we really need this check? `c.Generation != oldCluster.Generation`
We already have it in `SetupWithManager` and there will be no update in false case. 
And check all such cases
`	if event.ObjectNew.GetGeneration() == event.ObjectOld.GetGeneration() {
					return false
				}`

Reply:
The following logic prohibits updating the resource spec field. Validating webhooks trigger to update requests only on `Patch` and `Update` operation. However, we need to prevent any updations of the `spec`, so we have to understand what triggered an update. If it changes the `spec` it also changes the `metadata.generation` field. If we don't ensure that this is a `spec` update then it will denies any `Update` requests, but controllers of the following resources also update resources by adding/deleting annotations/finalizers by using `client.Update` and `client.Patch`.

In summary, It helps to understand if it was an update of the resource `spec` field or any other update, but we have to prohibit only updates of the `spec`.

Reply:
Validating webhook can validate the request and potentially reject it based on our business logic.
During this phase, the Generation field of the resource has not yet been incremented because the update has not been persisted. So, why should we check this?

Reply:
For the incoming new object it is already changed, because this is a part of it and it is changes only by k8s

Reply:
Ok